contracts/bitizens/AvatarItemToken.sol

```diff
@@ -6,218 +6,280 @@ import "./AvatarItemService.sol";
+    return itemTransferCount[_tokenId];
+  }
+
+  function getMineTrack(uint256 _tokenId) external view validToken(_tokenId) returns (uint256[]) {
+    return mineTrack[_tokenId];
+  }
 
   function getOwnedTokenIds(address _owner) external view onlyOperator returns(uint256[] _tokenIds) {
-    
+    require(_owner != address(0), "address invalid");
     return ownedTokens[_owner];
   }
 
-  function getAvatarItemInfo(uint256 _tokenId)
-    external
+  function getTokenInfo(uint256 _tokenId)
+    external 
     view 
-    returns(
-    uint32 _listNumber,
-    uint16 _setNumber,
-    bool _isBitizenItem, 
-    bool _isSet, 
-    uint8[6] _attr
-    ){
-    (_listNumber,_setNumber,_isBitizenItem,_isSet,_attr) = _getAvatarItemInfo(_tokenId);
+    validToken(_tokenId)
+    returns(address, address, bool, int16, uint256[4] _attr1, uint8[5] _attr2) {
+    Item storage item = tokenInfo[_tokenId];
+    _attr1[0] = item.node;
+    _attr1[1] = item.listNumber;
+    _attr1[2] = item.setNumber;
+    _attr1[3] = item.quality;  
+    _attr2[0] = item.rarity;
+    _attr2[1] = item.socket;
+    _attr2[2] = item.gender;
+    _attr2[3] = item.energy;
+    _attr2[4] = item.ext;
+    return (
+      item.foundedBy,
+      item.createdBy,
+      item.isBitizenItem,
+      item.probability,
+      _attr1,
+      _attr2
+      );
   }
 
-  function getBurnedTokenInfo(uint256 _tokenId) external view returns(  
-    uint32 _listNumber,
-    uint16 _setNumber,
-    bool _isBitizenItem, 
-    bool _isSet, 
-    uint8[6] _attr){
-    require(isBurnByTokenId[_tokenId]);
-    AvatarItem storage item = burnTokens[_tokenId];
-    _listNumber = item.listNumber;
-    _setNumber = item.setNumber;
-    _isBitizenItem = item.isBitizenItem;
-    _isSet = item.isSet;
-    _attr[0] = item.types;
-    _attr[1] = item.rarity;
-    _attr[2] = item.socket;      
-    _attr[3] = item.gender;
-    _attr[4] = item.ext1;
-    _attr[5] = item.ext2;  
+  function isBurned(uint256 _tokenId) external view validToken(_tokenId) returns (bool) {
+    return isBurnedToken[_tokenId];
+  }
+
+  function getBurnedTokenCount() external view returns (uint256) {
+    return burnedTokenIds.length;
+  }
+
+  function getBurnedTokenIdByIndex(uint256 _index) external view returns (uint256) {
+    require(_index < burnedTokenIds.length, "out of boundary");
+    return burnedTokenIds[_index];
+  }
+  
+  function getSameItemTokenIds(uint256 _tokenId) external view validToken(_tokenId) returns(uint256[]) {
+    return sameItemTokenIds[_getItemHash(_tokenId)];
   }
 
-  function isBurned(uint256 _tokenId) external view returns (bool) {
-    return isBurnByTokenId[_tokenId];
+  function getItemHash(uint256 _tokenId) external view validToken(_tokenId) returns (bytes32) {
+    return _getItemHash(_tokenId);
   }
 
-  function getBurnTokenCount() external view returns (uint256) {
-    return burnTokenIds.length;
+  function isSameItem(uint256 _tokenId1, uint256 _tokenId2)
+  external
```

Comment:
indentation. should have two leading spaces to be consistent

Reply:
Yes, it should have two leading space